lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AccessTokenEntity, ICrypto, IdTokenEntity, Logger, ScopeSet, Authority, AuthorityOptions, ServerAuthorizationROPCResponse, AccountEntity, AuthToken } from "@azure/msal-common";
+import { BrowserConfiguration, buildConfiguration, Configuration } from "../config/Configuration";
+import { SilentRequest } from "../request/SilentRequest";
+import { BrowserCacheManager } from "./BrowserCacheManager";
+import { ITokenCache } from "./ITokenCache";
+import { BrowserAuthError } from "../error/BrowserAuthError";
+
+export type LoadTokenOptions = {
+    clientInfo?: string,
+    extendedExpiresOn?: number
+};
+
+/**
+ * Token cache manager
+ */
+export class TokenCache implements ITokenCache {
+    // Flag to indicate if in browser environment
+    public isBrowserEnvironment: boolean;
+    // Input configuration by developer/user
+    protected config: BrowserConfiguration;
+    // Browser cache storage
+    private storage: BrowserCacheManager;
+    // Logger
+    private logger: Logger;
+    // Crypto class
+    private cryptoObj: ICrypto;
+
+    constructor(configuration: Configuration, storage: BrowserCacheManager, logger: Logger, cryptoObj: ICrypto) {
+        this.isBrowserEnvironment = typeof window !== "undefined";
+        this.config = buildConfiguration(configuration, this.isBrowserEnvironment);
```

Comment:
I don't think you need to rebuild the configuration, right? The config you pass in should already be type `BrowserConfiguration`

Reply:
You're right. Changed.